benchmark/benchmark_log.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <sstream>
+#include <string>
+
+#include "console_bridge/console.h"
+
+#include "performance_test_fixture/performance_test_fixture.hpp"
+
+using performance_test_fixture::PerformanceTest;
+
+class OutputHandlerString : public console_bridge::OutputHandler
+{
+public:
+  OutputHandlerString()
+  {
+  }
+
+  ~OutputHandlerString() override
+  {
+  }
+
+  void log(
+    const std::string & text, console_bridge::LogLevel level, const char * filename,
+    int line) override
+  {
+    (void)line;
+    (void)filename;
+    text_ = text;
+    log_level_ = level;
+  }
+
+  std::string text_;
+  console_bridge::LogLevel log_level_{console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_NONE};
+};
+
+class PerformanceFileHandlerTest : public performance_test_fixture::PerformanceTest
```

Comment:
Pretty sure you can drop the namespace because of the `using` statement at the beginning (which, for context, I found necessary for Google Benchmark to use it as a fixture).

If you think it makes it clearer to leave it in on purpose, I'm OK with that too. Same goes for in `SetUp` and `TearDown`.

```suggestion
class PerformanceFileHandlerTest : public PerformanceTest
```